hell/adventcalendar/2023/18.md

```diff
@@ -25,63 +25,100 @@ status:
 2. Tests have access to the real DOM, unlocking the ability to write tests for HTML, accessibility, and even CSS
-3. Tests can make use of native (not faked) interaction events, such as mouse clicks and keyboard input
+3. Tests can send native interactions, such as actual mouse clicks and keyboard input, instead of faking them with DOM events
 4. And it's even possible do things like set the viewport dimensions to test responsive design
 
 <!-- Manuel: "of native (not faked) interaction events" <- the difference needs explaining. -->
+<!-- David: I added some clarification around this, but it's a nuanced topic. Playwright has direct access to the browser's input pipeline (perform a mouse click at this coordinate in the window), whereas other testing libraries send DOM events that you'd _expect_ to fire _after_ an interaction like a mouse click occurs. There's a risk of these manually composed DOM events not matching the actual browser's behavior. Per my response to Eric above, I can cut this comparison if it's too much to discuss. -->
 
 <!-- Eric: I just want to take this space for communicating my appreciation of the use of an emdash 😃 -->
+<!-- David: I'm glad you're a fellow emdash fan! -->
 
 ## Shift accessibility feedback earlier
 
 With this approach, I write tests to capture my expectations around semantic markup and accessibility upfront, then get instant feedback on whether my code meets those expectations.
 
-<!-- Eric: A short paragraph here that describes what those expecations are and how they translate to code might be helpful here. -->
+Let's say I expect paragraph (`<p>`) elements shouldn't be empty. I can write the following test, which gets all `<p>` elements, loops over them, and expects each one doesn't have empty text content:
+```js
+describe("paragraphs", () => {
+	const docAllParagraphs = Array.from(document.querySelectorAll("img"));
+
+	it("have text content", () => {
+		docAllParagraphs.forEach((paragraph) => {
+			expect(paragraph.textContent).to.not.equal("");
+		});
+	});
+});
+```
+
+<!-- Eric: A short paragraph here that describes what those expectations are and how they translate to code might be helpful here. -->
+<!-- David: Good call. I added a simple example to prime readers on how this works. -->
 
 With access to the real DOM in tests, I can continuously check if I'm using heading levels, ARIA, and landmark regions correctly. I can build custom UI patterns and ensure they respond to keyboard input and manage focus as expected. And as I add new features or refactor my code, my tests re-run with every change to make sure I don't harm accessibility as the project evolves.
 
 <!-- Eric: I'm curious now about what harming accessibility might look like from a mature TDD-mindset. -->
+<!-- David: I'm not sure I understand "harming accessibility". Do you mean is it possible for developers who follow TDD to creating accessibility issues in their code? -->
```

Comment:
You wrote "sure I don't harm accessibility as the" in the paragraph and Eric wants to know what exactly harming means. :) But imo, we can leaves this paragraph as it is.

Reply:
Ah! Missed that. Thanks for the clarification.